python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -583,6 +583,7 @@ def run_request_for_partition(
         instance: Optional["DagsterInstance"] = None,
         current_time: Optional[datetime] = None,
     ) -> RunRequest:
+        # TODO deprecate this function?
```

Comment:
Hmm yes, I think we can deprecate this. Any downside you can think of? If we deprecate, we should explain the alternative in the docstring.

Reply:
I can't think of any downsides for doing this, I think it's generally much clearer to directly instantiate `RunRequest(partition_key=...)` and provides fewer entry points for us to manage